src/business_logic/transaction/objects/internal_invoke_function.rs

```diff
@@ -125,7 +126,7 @@ impl InternalInvokeFunction {
         // Execute transaction
         let call_info =
             self.run_execute_entrypoint(state, general_config, &mut resources_manager)?;
-        let updates_tracker_state = UpdatesTrackerState::new(state.clone());
+        let updates_tracker_state = CachedState::new(state.clone(), contract_classes);
```

Comment:
Instead of creating a new CachedState, use the state that the function receives in the inputs

Reply:
I think this fixes it: https://github.com/lambdaclass/starknet_in_rust/pull/143/commits/ceb8929965b652665acfc61262ef23da0332a8f3